### PR TITLE
enhancement(CM): bulk delete and unpublish migration to v5

### PR DESCRIPTION
**Pull request by madhurisandbhor**
### What does it do?

Migrating bulk delete and unpublish to V5

### How to test it?

`Bulk Delete` and `Bulk Unpublish` are in action on V5, no V5 design changes are included.

### Related issue(s)/PR(s)
⚠️ Merge after this related BE PR, https://github.com/strapi/strapi/pull/20184

### CODE REVIEW

File: packages/core/content-manager/admin/src/content-manager.ts [hunk 1]
```diff
@@ -63,7 +63,7 @@ interface ListViewContext {
   /**
    * TODO: add comment
```
Comment (joshuaellis) [inline packages/core/content-manager/admin/src/content-manager.ts:64]: Wanna add the comment? 😄

File: packages/core/content-manager/admin/src/hooks/useDocumentActions.ts [hunk 2]
```diff
@@ -29,20 +31,27 @@ import type {
...
 const DEFAULT_UNEXPECTED_ERROR_MSG = {
   id: 'notification.error',
   defaultMessage: 'An error occurred, please try again',
 } satisfies MessageDescriptor;
 
-type OperationResponse<TResponse extends { data: any; meta: any; error?: any }> =
+type OperationResponse<TResponse extends { data: any; meta?: any; error?: any }> =
   | Pick<TResponse, 'data' | 'meta'>
+  | { error: BaseQueryError | SerializedError }
+  | Pick<TResponse, 'data'>;
```
Comment (joshuaellis) [inline packages/core/content-manager/admin/src/hooks/useDocumentActions.ts:51]: can we potentially undo the changes here then?

File: packages/core/content-manager/admin/src/pages/ListView/components/BulkActions/Actions.tsx [hunk 3]
```diff
@@ -292,195 +299,146 @@ const BulkActionModal = ({
...
+  const { deleteMany: bulkDeleteAction } = useDocumentActions();
+  const documentIds = documents.map(({ documentId }) => documentId);
+
+  const handleConfirmBulkDelete = async () => {
+    const res = await bulkDeleteAction({
+      documentIds,
+      model,
+      params: {
+        locale: currentLocale,
+      },
```
Comment (joshuaellis) [inline packages/core/content-manager/admin/src/pages/ListView/components/BulkActions/Actions.tsx:319]: Pass all the params that exist, you can see an example here:

https://github.com/strapi/strapi/blob/94f313a12ead5a0e467b870efd12f7e2a9d30531/packages/core/content-manager/admin/src/pages/EditView/components/DocumentActions.tsx#L507

File: packages/core/content-manager/admin/src/pages/ListView/components/BulkActions/Actions.tsx [hunk 4]
```diff
@@ -292,195 +299,146 @@ const BulkActionModal = ({
...
+  const { formatMessage } = useIntl();
+  const { schema } = useDoc();
+  const { selectRow, selectedRows } = useTable('unpublishAction', (state) => state);
+  const hasPublishPermission = useDocumentRBAC('unpublishAction', (state) => state.canPublish);
+  const hasI18nEnabled = Boolean(schema?.pluginOptions?.i18n);
+  const { unpublishMany: bulkUnpublishAction } = useDocumentActions();
+  const documentIds = documents.map(({ documentId }) => documentId);
+
+  const handleConfirmBulkUnpublish = async () => {
+    const data = await bulkUnpublishAction({ documentIds, model });
```
Comment (joshuaellis) [inline packages/core/content-manager/admin/src/pages/ListView/components/BulkActions/Actions.tsx:379]: We should send the params here too :)

**Comment by joshuaellis** [comment]
you're basically 99% there

File: packages/core/content-manager/admin/src/content-manager.ts [hunk 5]
```diff
@@ -61,9 +61,9 @@ interface ListViewContext {
    */
   collectionType: string;
   /**
-   * TODO: add comment
+   * These are currently selected entries.
```
Comment (joshuaellis) [inline packages/core/content-manager/admin/src/content-manager.ts:64]:
```suggestion
   * The current selected documents in the table
```

File: packages/core/content-manager/admin/src/hooks/useDocumentActions.ts [hunk 6]
```diff
@@ -29,22 +31,28 @@ import type {
...
+  BulkUnpublish,
 } from '../../../shared/contracts/collection-types';
 
 const DEFAULT_UNEXPECTED_ERROR_MSG = {
   id: 'notification.error',
   defaultMessage: 'An error occurred, please try again',
 } satisfies MessageDescriptor;
 
-type OperationResponse<TResponse extends { data: any; meta: any; error?: any }> =
+type OperationResponse<TResponse extends { data: any; meta?: any; error?: any }> =
```
Comment (joshuaellis) [inline packages/core/content-manager/admin/src/hooks/useDocumentActions.ts:48]:
```suggestion
type OperationResponse<TResponse extends { data: any; meta: any; error?: any }> =
```

File: packages/core/content-manager/admin/src/pages/ListView/components/BulkActions/Actions.tsx [hunk 7]
```diff
@@ -292,195 +298,146 @@ const BulkActionModal = ({
...
-//         defaultMessage: 'Unpublish',
-//       }),
-//       onConfirm: handleConfirmUnpublishAllData,
-//     },
-//   };
-// };
+const DeleteAction: BulkActionComponent = ({ documents, model }) => {
+  const { formatMessage } = useIntl();
+  const { schema: contentType } = useDoc();
+  const { selectRow } = useTable('deleteAction', (state) => state);
```
Comment (joshuaellis) [inline packages/core/content-manager/admin/src/pages/ListView/components/BulkActions/Actions.tsx:304]:
```suggestion
  const selectRow = useTable('DeleteAction', (state) => state.selectRow);
```

Try and select just what you want, the selector calculates when it needs to re-render so it's just a bit better for performance

File: packages/core/content-manager/admin/src/pages/ListView/components/BulkActions/Actions.tsx [hunk 8]
```diff
@@ -292,195 +298,146 @@ const BulkActionModal = ({
...
+      ),
+      onConfirm: handleConfirmBulkDelete,
+    },
+  };
+};
+
+const UnpublishAction: BulkActionComponent = ({ documents, model }) => {
+  const { formatMessage } = useIntl();
+  const { schema } = useDoc();
+  const { selectRow, selectedRows } = useTable('unpublishAction', (state) => state);
```
Comment (joshuaellis) [inline packages/core/content-manager/admin/src/pages/ListView/components/BulkActions/Actions.tsx:369]: Do we need to be getting `selectedRows` when we have `documents` above?

File: packages/core/content-manager/admin/src/pages/ListView/components/BulkActions/Actions.tsx [hunk 9]
```diff
@@ -292,195 +298,146 @@ const BulkActionModal = ({
...
+            </Box>
+          )}
+        </Flex>
+      ),
+      onConfirm: handleConfirmBulkDelete,
+    },
+  };
+};
+
+const UnpublishAction: BulkActionComponent = ({ documents, model }) => {
```
Comment (joshuaellis) [inline packages/core/content-manager/admin/src/pages/ListView/components/BulkActions/Actions.tsx:366]: Shall we add a `type` so people can easily identify them in the reducer? i.e.

```
UnpublishAction.type = "unpublish"
```

**Review by joshuaellis — COMMENTED** [review]
minor polishes, but looks G R E A T 

Happy Birthday Bon Anniversaire GIF

File: packages/core/content-manager/admin/src/pages/ListView/components/BulkActions/Actions.tsx (hunk 9 above)
Comment (madhurisandbhor) [reply packages/core/content-manager/admin/src/pages/ListView/components/BulkActions/Actions.tsx:366]: I added actionType to indentify, as type was used [here](https://github.com/strapi/strapi/blob/fix/bulk-actions-migration/packages/core/content-manager/admin/src/pages/ListView/components/BulkActions/Actions.tsx#L44), couldn't omit it correctly

Comment (joshuaellis) [reply packages/core/content-manager/admin/src/pages/ListView/components/BulkActions/Actions.tsx:366]: I think there was some confusion how to approach this, check this PR https://github.com/strapi/strapi/pull/20256

It's done the same way we do document actions

**Review by joshuaellis — APPROVED** [review]
